main/src/ca/uwaterloo/flix/language/errors/ResolutionError.scala

```diff
@@ -1108,6 +1108,82 @@ object ResolutionError {
     override def explain(formatter: Formatter): Option[String] = None
   }
 
+  /**
+    * An error raised to indicate a struct does not exist in a `new Struct { ... } @ r` expression
+    *
+    * @param struct the names of the struct
+    * @param loc the location where the error occurred.
+    */
+  case class NonExistentStruct(struct: String, loc: SourceLocation) extends ResolutionError with Recoverable {
+    override def summary: String = s"Nonexistent struct type"
+
+    def message(formatter: Formatter): String = {
+      import formatter._
+      s""">> A struct type with the name `${struct}` does not exist
+         |
+         |${code(loc, "nonexistent struct")}
+         |""".stripMargin
+    }
+  }
+
+  /**
+    * An error raised to indicate a struct is missing a required field in `struct.field` or `struct.field = value` expression
+    *
+    * @param field the names of the missing fields
+    * @param loc the location where the error occurred.
+    */
+  case class NonExistentStructField(struct: String, field: String, loc: SourceLocation) extends ResolutionError with Recoverable {
+    override def summary: String = s"Struct $struct does not have the field $field"
+
+    def message(formatter: Formatter): String = {
+      import formatter._
+      s""">> Struct $struct does not have field $field
+         |
+         |${code(loc, "nonexistent field")}
+         |""".stripMargin
+    }
+  }
+
+  /**
+    * An error raised to indicate a `new` struct expression provides too many fields
+    *
+    * @param fields the names of the extra fields
+    * @param loc the location where the error occurred.
+    */
+  case class ExtraStructField(field: String, loc: SourceLocation) extends ResolutionError with Recoverable {
+    override def summary: String = s"`new` struct expression provides too many fields"
+
+    def message(formatter: Formatter): String = {
+      import formatter._
+      s""">> `new` struct expression provides field not present in original declaration of struct type
+         |
+         |${code(loc, "extra fields")}
+         |
+         |Extra Field: ${field}
+         |""".stripMargin
+    }
+  }
+
+  /**
+    * An error raised to indicate a `new` struct expression is missing fields
+    *
+    * @param fields the names of the missing fields
+    * @param loc the location where the error occurred.
+    */
+  case class UnprovidedStructField(field: String, loc: SourceLocation) extends ResolutionError with Recoverable {
```

Comment:
-> UndefinedStructField

Reply:
What's the two "uses" of UndefinedStructField?

Reply:
This one indicates that a `new` expression is missing a field

The other one indicates you tried to get a field that doesn't exist

Reply:
Ah okay. For the new one, lets call it missing struct field in initializer. For the other one, let us call it undefined field.

Reply:
Maybe that is already what you have?